lib/api/src/Cardano/Wallet/Api/Types/Error.hs

```diff
@@ -217,6 +217,7 @@ data ApiErrorInfo
     | BlockHeaderNotFound
     | TranslationByronTxOutInContext
     | BalanceTxInlinePlutusV3ScriptNotSupportedInBabbage
+    | UnsupportedEraInTxSubmission
```

Comment:
Two suggestions:

1. Perhaps we can shorten the name to `UnsupportedEra`?
    Justification: The context of the error is already clear from the name of the endpoint that returned it. If in future we wish to return a similar error from another endpoint, we can just reuse this one.

2. Perhaps consider making this into a rich error type?

Something like this:
```suggestion
    | UnsupportedEra !ApiErrorUnsupportedEra
        
data ApiErrorUnsupportedEra = ApiErrorUnsupportedEra
    { unsupportedEra :: !ApiEra
    -- ^ The unsupported era (as specified by the caller).
    , supportedEras :: !(Set ApiEra)
    -- ^ The set of eras that we currently support.
    }
    deriving (Data, Eq, Generic, Show, Typeable)
    deriving (FromJSON, ToJSON) via DefaultRecord ApiErrorUnsupportedEra
    deriving anyclass NFData
```